File: java/codecheck/Args.java
package codecheck;

public class Args {
	
	static int exnumber;
	static int array_len;
	static char[] array;

	Args(){
		exnumber = -1;
	}

	public static boolean invalid(char c){

		if(exnumber == 1){
			return (c < '0' || c > '9');
		}
		else if(exnumber == 2){
			return (c < '0' || c > '1');
		}
		else if(exnumber == 3){
			return (c < 'A' || (c > 'Z' && c < 'a') || c > 'z');
		}
		else{
			return true;
		}
			
	}
	
	public static void parseArgs(int argc, String[] argv){

		if (argc < 2){
	        System.err.println("");
			System.exit(1);
		}
		
		try{
			exnumber = Integer.parseInt(argv[0]);
			if(exnumber < 1 || exnumber > 3){
				throw new java.lang.NumberFormatException("");
			}
		}
		catch(java.lang.NumberFormatException e){
            System.err.println("");
			System.exit(1);
		}

		array_len = argv[1].length();
		array = argv[1].toCharArray();
		
		for(int i=0; i<array_len; i++){
			if(invalid(array[i])){
                System.err.println("");
				System.exit(1);
			}
		}
	}	
}

File: c++/src/args.cc
#include <stdlib.h>
#include <string.h>
#include <iostream>

#include "args.h"

namespace exercise{
  
  Args::Args(){
	exnumber = -1;
	array = "";
  }
  
  void Args::parseArgs(int argc, char** argv){

	if(argc != 3){
	  //std::cout << "Expect two arugments" << std::endl;
	  printHelp();
	  exit(EXIT_FAILURE);
	}
	
	std::string str_exnumber = argv[1];
	array = argv[2];
	
	try{
	  exnumber = stoi(str_exnumber);
	  if(exnumber < 1 || exnumber > 3){
		throw std::invalid_argument("Invalid number");
	  }
	}
	catch(std::invalid_argument e){
	  //std::cout << "Expect first arugment: 0-4" << std::endl;  
	  exit(EXIT_FAILURE); 
	}

	std::string expect;
	std::function<bool(char)> invalid;
	
	if(exnumber == 1){
	  expect = "number";
	  invalid = [=](char c) -> bool {return (c < '0' || c > '9');};
	}
	else if(exnumber == 2){
	  expect = "binary";
	  invalid = [=](char c) -> bool {return (c < '0' || c > '1');};
	}
	else if(exnumber == 3){
	  expect = "a-z or A-Z";
	  invalid = [=](char c) -> bool {return (c < 'A' || (c > 'Z' && c < 'a') || c > 'z');};
	}
	
	try{
	  // if(array == ""){
	  // 	throw std::invalid_argument("Invalid char");
	  // }
	  
	  for(std::string::iterator it=array.begin(); it!=array.end(); ++it){
		if(invalid(*it)){
		  throw std::invalid_argument("Invalid char");
		}
	  }
	}
	catch(std::invalid_argument e){
	  //std::cout << "Expect array arugment all char: " << expect << std::endl;  
	  exit(EXIT_FAILURE); 
	}
  }
  
  void Args::printHelp(){
	std::cout
	  << "Usage\n"
	  << "  first arg;             exercise number\n"
	  << "  array arg:             input array\n"
	  << std::endl;
  }
}
